src/credentials/CertificationDeclaration.cpp

```diff
@@ -170,6 +183,22 @@ CHIP_ERROR DecodeCertificationElements(const ByteSpan & encodedCertElements, Cer
     }
     VerifyOrReturnError(err == CHIP_END_OF_TLV || err == CHIP_ERROR_UNEXPECTED_TLV_ELEMENT || err == CHIP_NO_ERROR, err);
 
+    if (reader.GetTag() == ContextTag(kTag_AuthorizedPAAList) && reader.GetType() == kTLVType_Array)
+    {
+        ReturnErrorOnFailure(reader.EnterContainer(outerContainer2));
+
+        certElements.AuthorizedPAAListCount = 0;
+        while ((err = reader.Next(AnonymousTag())) == CHIP_NO_ERROR)
+        {
+            ReturnErrorOnFailure(
+                reader.GetBytes(certElements.AuthorizedPAAList[certElements.AuthorizedPAAListCount++], kKeyIdentifierLength));
+        }
+        VerifyOrReturnError(err == CHIP_END_OF_TLV, err);
+        ReturnErrorOnFailure(reader.ExitContainer(outerContainer2));
+    }
+    err = reader.Next();
```

Comment:
Why is this outside the paa list block?  I'd expect it to be inside.

Reply:
Agree, that belongs to the logic inside the block. Updated